src/metaopt/io/database/__init__.py

```diff
@@ -20,7 +20,7 @@
 
 import six
 
-from metaopt.utils import AbstractSingletonType
+from metaopt.utils import (AbstractSingletonType, SingletonFactory)
```

Comment:
What is `SingletonFactory`?

Reply:
`Factory` is a metaclass to create the `Database` class, a factory pattern aggregating within its creation both the argument denoting its implementation database type (see `of_type` argument), and the arguments to be passed for the creation/initialization of an implementation of the interface, defined by the base class of the Factory class to-be-created (in our case is the `AbstractDB`). See also the tests for how it is used.

`SingletonFactory` is the `Factory` metaclass wrapped by the `Singleton` metaclass, so that it is allowed to return only one instantiation of the created class.